test/unit/lib/buildHeaderItems.js

```diff
@@ -0,0 +1,84 @@
+const chai = require('chai');
+const buildHeaderItems = require('../../../app/lib/header/buildHeaderItems');
+const cleanUrl = require('../../../app/lib/header/cleanUrl');
+const headerApiResponse = require('../../resources/headerApiResponse');
+
+const expect = chai.expect;
+
+function createMenuItem(title, partUrl, submenus) {
+  return {
+    Title: title,
+    URL: `http://site/${partUrl},${title}`,
+    Submenus: submenus
+  };
+}
+
+describe('cleanUrl', () => {
+  it('should replace \'http://site\' with  \'https://www.nhs.uk\' and remove text after comma', () => {
+    const title = 'Health A-Z';
+    const partUrl = 'Conditions/Pages/hub.aspx';
+    const url = `http://site/${partUrl},${title}`;
+    const output = cleanUrl(url);
+    expect(output).to.equal(`https://www.nhs.uk/${partUrl}`);
+  });
+
+  it('should replaces spaces with \'%20\'', () => {
+    const output = cleanUrl('http://site/some url');
+    expect(output).to.equal('https://www.nhs.uk/some%20url');
+  });
+
+  it('should gracefully handle empty urls', () => {
+    const output = cleanUrl('');
+    expect(output).to.equal('');
+  });
+
+  it('should gracefully handle undefined urls', () => {
+    const output = cleanUrl(undefined);
+    expect(output).to.equal('');
+  });
+});
+
+describe('buildHeaderItems', () => {
```

Comment:
Can this be a in separate file to the execution of the tests for `cleanUrl`?

Reply:
will do, was in the same file as they are so closely related. Will also rename it 'cleanHeaderUrl to make it explicit that it's a url from the header api